### PR TITLE
Ernie image

### PR DESCRIPTION
**Pull request by mi804**
# Baidu/ERNIE-Image Integration

## Summary

Integrate [Baidu/ERNIE-Image](https://modelscope.cn/models/baidu/ERNIE-Image) into DiffSynth-Studio, supporting:

- ✅ Text-to-Image — Text prompt to high-quality image generation
- ✅ Text-to-Image with Prompt Enhancement — Optional PE module (Ministral3) that rewrites prompts for richer, more detailed outputs

This is a **new_series** integration. ERNIE-Image uses a custom SharedAdaLN + RoPE 3D + joint image-text attention architecture, distinct from existing DiT models (FLUX, Qwen-Image, Wan, etc.) in DiffSynth-Studio. The DiT is a 36-layer ~4.7B parameter transformer.

### Model Components

| Component | File | Lines | Params | Converter |
|-----------|------|-------|--------|-----------|
| DiT | `diffsynth/models/ernie_image_dit.py` | 367 | ~4.7B | `ErnieImageDiTStateDictConverter` |
| VAE | Reuses existing `Flux2VAE` (`diffsynth/models/flux2_vae.py`) | — | — | Reuses existing |
| Text Encoder | `diffsynth/models/ernie_image_text_encoder.py` | 77 | ~3.8B | `ErnieImageTextEncoderStateDictConverter` |
| PE | `diffsynth/models/ernie_image_pe.py` | 116 | ~3.8B | `ErnieImagePEStateDictConverter` |

**Pipeline**: `diffsynth/pipelines/ernie_image.py` (386 lines) — `ErnieImagePipeline`

### Dependencies Added

- `diffusers>=0.38.0.dev0` (optional, only for target library validation, not for DiffSynth runtime)
- `flash-attn` (optional, CUDA acceleration)
- All LLM components use **transformers** directly: `Ministral3Model` (text encoder), `Ministral3ForCausalLM` (PE). Weights loaded automatically via transformers.

#### Text-to-Image
- Standard text-to-image generation with FlowMatchScheduler (`ERNIE-Image` template)
- Default: 1024x1024, 50 steps, CFG=4.0, bfloat16
- Supports full training and LoRA training (training pipeline has I2I data path)

#### Text-to-Image with Prompt Enhancement
- Optional `use_pe=True` flag enables prompt rewriting via Ministral3 Causal LM
- Returns both generated image and the revised prompt
- PE temperature and top_p configurable (`pe_temperature=0.6`, `pe_top_p=0.95` defaults)
- PE module is entirely optional — disabling it has no impact on base generation
- No training support for PE (inference only)

### Example Usage

```python
from diffsynth.pipelines.ernie_image import ErnieImagePipeline, ModelConfig

# Basic Text-to-Image
pipe = ErnieImagePipeline.from_pretrained(
    torch_dtype=torch.bfloat16,
    model_configs=[
        ModelConfig(model_id="baidu/ERNIE-Image", origin_file_pattern="transformer/diffusion_pytorch_model*.safetensors"),
        ModelConfig(model_id="baidu/ERNIE-Image", origin_file_pattern="text_encoder/model.safetensors"),
        ModelConfig(model_id="baidu/ERNIE-Image", origin_file_pattern="vae/diffusion_pytorch_model.safetensors"),
    ],
)
image = pipe(prompt="一只黑白相间的中华田园犬", num_inference_steps=50, cfg_scale=4.0)
image.save("output.jpg")

# With Prompt Enhancement
pipe_pe = ErnieImagePipeline.from_pretrained(
    torch_dtype=torch.bfloat16,
    model_configs=[
        ModelConfig(model_id="baidu/ERNIE-Image", origin_file_pattern="transformer/diffusion_pytorch_model*.safetensors"),
        ModelConfig(model_id="baidu/ERNIE-Image", origin_file_pattern="text_encoder/model.safetensors"),
        ModelConfig(model_id="baidu/ERNIE-Image", origin_file_pattern="vae/diffusion_pytorch_model.safetensors"),
        ModelConfig(model_id="baidu/ERNIE-Image", origin_file_pattern="pe/model.safetensors"),
    ],
)
image, revised_prompt = pipe_pe(prompt="一只黑白相间的中华田园犬", use_pe=True)
print(f"Revised prompt: {revised_prompt}")
```

### Architecture Details

- **DiT**: SharedAdaLN structure with shift/scale/gate parameters shared between self-attention and MLP. RoPE 3D positional encoding across (t, y, x) dimensions. Joint attention over image tokens + text tokens. Internal format: `[S, B, H]` for transformer blocks, `[B, S, H]` for attention.
- **Text Encoder**: `Ministral3Model` (transformers), 26 layers, hidden_size=3072, yarn RoPE. Outputs `hidden_states[-2]` (second-to-last layer) as text embeddings.
- **PE**: `Ministral3ForCausalLM` (transformers), same config as text encoder. Uses chat_template for prompt rewriting, carries prompt text + target resolution in JSON format.
- **VAE**: Directly reuses `Flux2VAE` — all parameters match (block_out_channels, latent_channels=32, patch_size=(2,2), etc.).

### Scripts

| Script | Feature | Description |
|--------|---------|-------------|
| `examples/ernie_image/model_inference/Ernie-Image-T2I.py` | T2I | Base text-to-image (no PE) |
| `examples/ernie_image/model_inference/Ernie-Image-T2I-PE.py` | T2I + PE | Text-to-image with prompt enhancement |
| `examples/ernie_image/model_inference_low_vram/Ernie-Image-T2I.py` | T2I (low VRAM) | VRAM offload to CPU |
| `examples/ernie_image/model_inference_low_vram/Ernie-Image-T2I-PE.py` | T2I + PE (low VRAM) | VRAM offload + PE |
| `examples/ernie_image/model_training/train.py` | Training | Training framework for full + LoRA |
| `examples/ernie_image/model_training/full/Ernie-Image-T2I.sh` | Full training | DeepSpeed ZeRO-3 config |
| `examples/ernie_image/model_training/lora/Ernie-Image-T2I.sh` | LoRA training | LoRA training script |
| `examples/ernie_image/model_training/validate_full/Ernie-Image-T2I.py` | Full training validation | Load epoch-1 checkpoint, generate images |
| `examples/ernie_image/model_training/validate_lora/Ernie-Image-T2I.py` | LoRA training validation | Load epoch-4 LoRA checkpoint, generate images |

### Notes

- **Hardware**: Recommended 24GB+ VRAM for bfloat16 inference. Low VRAM configurations available with CPU offload (see `examples/ernie_image/model_inference_low_vram/`).
- **Scheduler**: Uses FlowMatchScheduler with `ERNIE-Image` template (exponential shift). Scheduler parameters are fixed and not exposed to the user.
- **VAE**: Reuses the existing `Flux2VAE` class — no new VAE code added.
- **Text Encoder & PE**: Both components wrap transformers Ministral3 models (not Mistral3). Weights are loaded automatically via transformers; custom converters exist for config mapping.
- **Training**: Supports full training (DeepSpeed ZeRO-3) and LoRA training. PE has no training support.
- **VRAM Management**: Pipeline supports VRAM offload via `vram_config` in ModelConfig and `vram_limit` parameter. DiT is the only model kept on-device during denoising iterations.
- **Tiled VAE**: Supports tiled VAE decode via `tiled=True` parameter for large images.

### CODE REVIEW

File: diffsynth/pipelines/ernie_image.py [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+from ..diffusion import FlowMatchScheduler
+from ..core import ModelConfig, gradient_checkpoint_forward
+from ..diffusion.base_pipeline import BasePipeline, PipelineUnit
+from ..models.ernie_image_text_encoder import ErnieImageTextEncoder
+from ..models.ernie_image_dit import ErnieImageDiT
+from ..models.ernie_image_pe import ErnieImagePE
+from ..models.flux2_vae import Flux2VAE
+
+
+# ============================================================
```
Comment (Artiprocher) [inline diffsynth/pipelines/ernie_image.py]: Unnecessary annotation. Please delete it.

File: diffsynth/pipelines/ernie_image.py [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+        super().__init__(
+            device=device, torch_dtype=torch_dtype,
+            height_division_factor=16, width_division_factor=16,
+        )
+        self.scheduler = FlowMatchScheduler("ERNIE-Image")
+        self.text_encoder: ErnieImageTextEncoder = None
+        self.dit: ErnieImageDiT = None
+        self.vae: Flux2VAE = None
+        self.tokenizer: AutoTokenizer = None
+        self.pe: ErnieImagePE = None
```
Comment (Artiprocher) [inline diffsynth/pipelines/ernie_image.py]: Prompt enhancement is not the capability of the base model. Please remove the prompt enhancement modules, including the model and the pipeline unit. We can assume that the inputed prompt is already enhanced by other tools outside.

File: diffsynth/utils/state_dict_converters/ernie_image_dit.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+def ErnieImageDiTStateDictConverter(state_dict):
```
Comment (Artiprocher) [inline diffsynth/utils/state_dict_converters/ernie_image_dit.py]: This StateDictConverter doesn't do anything. Please remove it.
